File: client/python/mesh/http/consumer.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#

from concurrent.futures import Future

from mesh.kinds import Reference
from mesh.macro import spi
from mesh.mpc import Consumer, Execution


@spi("http")
class HTTPConsumer(Consumer):

    def __enter__(self):
        pass

    def __exit__(self, exc_type, exc_val, exc_tb):
        pass

    def start(self):
        pass

    def consume(self, address: str, urn: str, execution: Execution[Reference], inbound: bytes) -> Future:
        pass

File: client/javascript/src/grpc/consumer.ts
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

import {Consumer, Execution} from "@/mpc";
import {spi} from "@/macro";
import {Context} from "@/prsim";
import {Reference} from "@/kinds";
import {Status} from "@/cause";

@spi(Consumer.GRPC)
export class GRPCConsumer extends Consumer {

    start(): void {

    }

    close(): void {
    }

    consume(ctx: Context, address: string, urn: string, execution: Execution<Reference>, inbound: Uint8Array): Promise<Uint8Array> {
        return Promise.reject(Status.URN_NOT_PERMIT);
    }

}